### DISCUSSION TITLE
How to get the server lists?

### DISCUSSION THREAD

**Discussion by jmhIcoding** [qdm12/gluetun#569] [Q&A]
I want to know how to get vpn server lists? thus I could update the server lists by myself.

**Comment by qdm12** [comment]
Hello, see https://github.com/qdm12/gluetun/wiki/Updating-Servers

**Reply by jmhIcoding** [reply]
> Hello, see https://github.com/qdm12/gluetun/wiki/Updating-Servers

Hello, I want to add SafeConnection VPN (The app could be found in Google Play Store https://play.google.com/store/apps/details?id=com.mcafee.safeconnect.android), because I have subscribed a VIP account for this VPN App. 
And my question is : how could I find out all the VPN nodes of this app?

**Reply by qdm12** [reply]
Oh in this case it's more complicated. You have two options:

- Create an issue [here](https://github.com/qdm12/gluetun/issues/new?assignees=&labels=%3Abulb%3A+New+provider&template=provider.md&title=VPN+provider+support%3A+NAME+OF+THE+PROVIDER) - note that it will take me a few weeks since there are already pending requests for several other vpn providers.
- Use an openvpn configuration file directly, see https://github.com/qdm12/gluetun/wiki/Openvpn-file

If you can only connect with their mobile app and can't find a way to get the openvpn files (or wireguard files in the future) then it won't work unfortunately.
